File: app/src/main/java/com/example/r_clazz/Adapters/Course_Adapter.java
package com.example.r_clazz.Adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.example.r_clazz.Been.Course_Been;
import com.example.r_clazz.R;

import java.util.List;

public class Course_Adapter extends ArrayAdapter<Course_Been> {

    private int resouseID;
    public Course_Adapter(@NonNull Context context, int resource, @NonNull List<Course_Been> objects) {
        super(context, resource, objects);
        this.resouseID = resource;
    }


    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        Course_Been course_been = getItem(position);
        View view = LayoutInflater.from(getContext()).inflate(resouseID,parent,false);
        TextView clazz_name = view.findViewById(R.id.course_title);
        TextView clazz_code = view.findViewById(R.id.codings);
        clazz_name.setText(course_been.getCourse_name());
        clazz_code.setText(course_been.getCourse_code());
        return view;
    }
}

File: app/src/main/java/com/example/r_class/model/adapters/CourseAdapter.kt
package com.example.r_class.model.adapters

import android.annotation.SuppressLint
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import com.example.r_class.R
import com.example.r_class.model.entity.Course

class CourseAdapter(val context: Context,val data:ArrayList<Course>) : BaseAdapter() {
    override fun getCount(): Int {
        return data.size
    }

    override fun getItem(position: Int): Any {
        return data[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }


    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
       var view : View
       var holder : CourseHolder

       if (convertView == null){
           view = LayoutInflater.from(context).inflate(R.layout.clazz_item,null)
           holder = CourseHolder()
           holder.courseName = view.findViewById(R.id.course_title)
           holder.courseCode = view.findViewById(R.id.codings)
           view.tag = holder
       }else{
           view = convertView
           holder = convertView.tag as CourseHolder
       }

        val item = getItem(position) as Course

        holder.courseCode?.text = item.course_code
        holder.courseName?.text = item.course_name

        return view
    }





}